install_requirements.py

```diff
@@ -0,0 +1,153 @@
+import os
+import platform
+import re
+import sys
+import subprocess
+
+# Before doing anything, cd to the directory containing this script.
+os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+# Find the names of the python tools to use.
+PYTHON_EXECUTABLE = os.getenv('PYTHON_EXECUTABLE')
+
+if not PYTHON_EXECUTABLE:
+    CONDA_DEFAULT_ENV = os.getenv('CONDA_DEFAULT_ENV')
+    if not CONDA_DEFAULT_ENV or CONDA_DEFAULT_ENV == "base" or not subprocess.call(["which", "python"]):
+        PYTHON_EXECUTABLE = "python3"
+    else:
+        PYTHON_EXECUTABLE = "python"
+
+if PYTHON_EXECUTABLE == "python":
+    PIP_EXECUTABLE = "pip"
+else:
+    PIP_EXECUTABLE = "pip3"
+
+print(f"Using Python executable: {PYTHON_EXECUTABLE}")
+print(f"Using Pip executable: {PIP_EXECUTABLE}")
+
+def python_is_compatible():
+    # Scrape the version range from pyproject.toml, which should be in the current directory.
+    version_specifier = None
+    with open('pyproject.toml', 'r') as file:
+        for line in file:
+            if line.startswith('requires-python'):
+                match = re.search(r'"([^"]*)"', line)
+                if match:
+                    version_specifier = match.group(1)
+                    break
+
+    if not version_specifier:
+        print("WARNING: Skipping python version check: version range not found", file=sys.stderr)
+        return False
+
+    # Install the packaging module if necessary.
+    try:
+        import packaging
+    except ImportError:
+        subprocess.check_call([PIP_EXECUTABLE, 'install', 'packaging'])
+    # Compare the current python version to the range in version_specifier. Exits
+    # with status 1 if the version is not compatible, or with status 0 if the
+    # version is compatible or the logic itself fails.
+    try:
+        import packaging.version
+        import packaging.specifiers
+
+        python_version = packaging.version.parse(platform.python_version())
+        version_range = packaging.specifiers.SpecifierSet(version_specifier)
+        if python_version not in version_range:
+            print(
+                f"ERROR: ExecuTorch does not support python version {python_version}: must satisfy \"{version_specifier}\"",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+    except Exception as e:
+        print(f"WARNING: Skipping python version check: {e}", file=sys.stderr)
+        sys.exit(0)
+        return False
+    return True
+
+if not python_is_compatible():
+    sys.exit(1)
```

Comment:
The caller of this code already does a sys.exit(1) when it fails, so it'd be better to only return true/false and leave the reaction to the caller. Also, calling `sys.exit(0)` would kill the whole script, but we want to continue in the "skipping check" cases.

The bash script called `sys.exit()` because the version-checking logic was run in a separate subshell.

```suggestion
        if python_version not in version_range:
            print(
                f"ERROR: ExecuTorch does not support python version {python_version}: must satisfy \"{version_specifier}\"",
                file=sys.stderr,
            )
            return False
    except Exception as e:
        print(f"WARNING: Skipping python version check: {e}", file=sys.stderr)
    return True

if not python_is_compatible():
    sys.exit(1)
```